stdlib.sh

```diff
@@ -427,6 +427,26 @@ layout_python3() {
   layout_python python3 "$@"
 }
 
+# Usage: layout pipenv
+#
+# Similar to layout_python, but uses Pipenv to build a
+# virtualenv from the Pipfile located in the same directory.
+#
+layout_pipenv() {
+  if [[ ! -f Pipfile ]]; then
+    echo 'No Pipfile found.  Use `pipenv` to create a Pipfile first.' >&2
```

Comment:
is there a `pipenv init` that could make sense here?

Reply:
That command does not exist. I'm not sure if invoking that would work here, since it would have to decide between Python 2 or Python 3 on the user's behalf.

Do you think dropping that message and then returning with a 0 exit code is better? I noticed that direnv keeps throwing the error after every command you type if you're stuck in this state.

Reply:
No that's good, I just wanted to check if there is anything helpful that we could do instead of returning an error. Since it's not the case the error is good.

Actually you should use `log_error` to respect the direnv log format.

The error after every command is a but that I meant to fix. It should only fail once and stay in the errored state.

Reply:
Pushed the change for the error logging in 4748fbe

Reply:
Oh, much nicer — TIL about `log_error`.

Reply:
👍